processing/src/main/java/io/druid/segment/DimensionSelector.java

```diff
@@ -117,4 +117,34 @@
    */
   @Nullable
   IdLookup idLookup();
+
+  /**
+   * @deprecated always throws {@link UnsupportedOperationException}
+   */
+  @Deprecated
+  @Override
+  default float getFloat()
```

Comment:
if this is suppose to be always the case why not making this as final methods instead of defaults ?